### ISSUE TITLE
A11y documentation audit: Review Design section documentation

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/va.gov-team#35946]
## Description
We want to audit our accessibility documentation on design.va.gov to make sure that it is up-to-date with current accessibility guidelines.

For this ticket, we need to review the accessibility guidelines within the Design section of design.va.gov. Specifically, the guidelines on these three pages:
- https://design.va.gov/design/
- https://design.va.gov/design/icons
- https://design.va.gov/design/principles

We would like for you to review the specified pages and provide any updates that need to be made to the existing documentation as a comment on this ticket.

## Details
Just for reference, here are some useful links:
[Platform accessibility standards](https://vfs.atlassian.net/wiki/spaces/pilot/pages/1777893393/WCAG+2.1+success+criteria+and+foundational+testing)
[VFS Team accessibility responsibilities](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/roles-and-responsibilities/accessibility.md#accessibility)
[VA.gov experience standards](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)
VSP WCAG 2.1 Presentations:
[VSP WCAG2.1 Foundational Testing Redux](https://app.zenhub.com/files/133843125/368da85e-7f1f-4332-8ccb-6808e52a8a51/download)
[VSP WCAG2.1 New Success Criteria](https://app.zenhub.com/files/133843125/5475fd9b-9d2e-4a6a-901f-97682664eb33/download)

## Acceptance Criteria
- [ ] Review accessibility guidelines on the following pages of design.va.gov:
    - [ ] https://design.va.gov/design/
    - [ ] https://design.va.gov/design/icons
    - [ ] https://design.va.gov/design/principles
- [ ] Add notes on any necessary updates as a comment on this ticket
- [ ] Ping Katy, Carol, and Natalie for review

**Comment by CourtneyChambers77** [comment]
Address block or Address Information
Accessibility:

Some Criteria on Address info: 
•	Contact information can be email address, physical location, website address, phone number. Address block cannot be used for arbitrary addresses.
•	When author uses address element for such purposes then they may think it provides certain semantical information to assistive technology users. 
•	Voice over on iPhone/mac conveys as content information landmark semantics when address tag is used, and this is completely incorrect announcement/semantics. 
•	Adding a landmark elements and roles, such as <aside> or role="contentinfo, to demarcate page regions as assistive technology allow the user to quickly navigate to these sections.
•	To ensure the screen reader is giving this information to the user add an aria-label="  ”. Aria-label is a description that is never displayed on screen but is relayed to the screen reader user. This can be very helpful if you have multiple navigation areas within a single page.

Accessibility considerations for breadcrumb
•	The breadcrumb should remain in the same relative order and location on each web page.
•	The link text should be clear, reflect the breadcrumb location accurately and easy to understand for all users.
•	The link to the current page element in the breadcrumb has the aria-current="page" attribute.

Accessibility considerations for button
•	When using the <button> element, always specify a type.
•	Buttons should display a visible focus state when users tab to them.
•	The button should have text or label that accurately describes the action of the button.
•	Avoid using <div> or <img> tags to create buttons. Screen readers don’t automatically know either is a usable button.
•	Include more contextual information in the button label for screen readers. You can use an aria label to specify form numbers or program names in the buttons for greater context.
•	If a description of the button’s function is present, the button element should have an aria-describedby set to the ID of the element containing the description.
•	It is important to use Action links for calls to actions that link to another page rather than buttons, because screen readers always say “link” before links, and “button” before buttons.
. 

•	Button and link confusion can be very frustrating for assistive technology users. A user with a screen reader may pull up a list of links and may not find a specific link because it turns out that it has actually been designated as a button in the markup.
•	Pressing the Space key triggers a button, but pressing the Enter key triggers a link.
•	Using buttons and links intentionally results in a more inclusive experience for assistive technology users. Make sure to read both button and action link guidance to determine what is needed for a page.
Button Groups
•	Don’t mix differently-sized buttons (like a default button and a big button) in the same group.
•	Use short, descriptive button labels to describe controls.
•	A current state and outline buttons as a non-current state. The segmented button group works with all button variants. However, you may need to create custom styles to distinguish state for some button variants.
Accessibility considerations form 
•	Provide labels and instructions that are clear and concise.
•	Avoid placeholder text for accessibility reasons. Most browsers’ default rendering of placeholder text does not provide a high enough contrast ratio.
•	Avoid breaking numbers with distinct sections (such as phone numbers, Social Security Numbers, or credit card numbers) into separate input fields. For example, use one input for phone number, not three (one for area code, one for local code, and one for number). Each field needs to be labeled for a screen reader and the labels for fields broken into segments are often not meaningful.
•	When the input is a required field include the aria-required property and indicate that it is a required field. 
•	Voiceover on iOS currently does not support fieldset and legend for forms. You can address this by using aria-labelledby="for-attribute-of-label id-of-legend id-of-additional-info" on each input in the fieldset. Using aria-labelledby will overwrite the default text read by the screen reader, so it is important to include all relevant information.
•	Voiceover on OS X offers partial support for aria-describedby. There's full support for conveying description on text inputs, partial support for description changes when the text input is in focus, and no support for role="alert".
Accessibility considerations Modal 
•	Modals should have a simple headline that explains its purpose. 
•	Use enough descriptive text to be clear what the user needs to do and why. 
•	Avoid using modals to display complex forms or large amounts of information.
•	Avoid complicated user flows in a modal that may take the user away from the original page. 
•	A multi-step process is better suited to an individual page, guiding the user and accommodating complexities in the user flow.
•	When using a form field, these are better displayed in the context of the page (such as showing an error next to a missing required field). 
•	When these relate to page-level messages (such as confirming that a form has been successfully submitted), they should appear as an alert at the top of the page where the user is taken next.
Accessibility considerations Pagination 
•	Paginated content should be split into multiple pages determined only by a specific amount of content per page, not split by any meaningful attribute, like feature or subject or step. 
•	Readers use the pagination component to move from page to page in paginated content, or directly to the first or last page of the paginated set.
•	Users should always be able to navigate to each of these pages from any page in the set.
•	Use a wrapping <nav> element to identify Pagination as a navigation section to screen readers and other assistive technologies.
•	Use a unique, descriptive ARIA label for the <nav> element.
•	Use an unordered list for the navigation items. This allows screen readers users to voice the number of elements in the Pagination component
Accessibility considerations Progress bar 
•	Progress indicators are key for visibility of system status. 
•	For assistive technology user the visually represent a path to completion of a particular task or process. 
•	The indicators help the user identify how much of the process the user has completed, and how much is still left.
•	The visually represent a page or element that is loading, use a spinner or progress bar rather than a steps indicator.
•	Progress bars show determinate levels of progress, meaning there is a clear point of completion. When the progress bar is filled and hits 100%, the application should complete the process.
•	Ensure that information denoted by the color is either obvious from the content itself (visible text), or is included through alternative means, such as additional text hidden with the Screen Reader (SR) only class.

**Comment by k80bowman** [comment]
@CourtneyChambers77 this is all good info, but this ticket is only for the pages specified and linked to in the description and Acceptance Criteria above. Are there any changes you would recommend for the accessibility guidance on those three pages?

**Comment by CourtneyChambers77** [comment]
I agree a lot of these components have no variants. Tracking the accessibility consideration leads into recommendation very thin line I do agree with you

**Comment by CourtneyChambers77** [comment]
https://design.va.gov/design/icons
Accessibility
When using an icon that's decorative shouldn’t be read out to users because it doesn’t add to their understanding.
When icons are used to illustrate items, people with disabilities such as a blind person, will not be able to comprehend. Icons should be accompanied with text. 
All interactive accessible icons must be keyboard accessible and touch accessible as well as clickable via a mouse.

**Comment by CourtneyChambers77** [comment]
https://design.va.gov/design/principles
VA.gov design principles
VA.gov design principles are fundamental pieces of guidance that help designers make their applications consistent with DEPO’s standards of quality in the user experience. Designers should apply them when they conceive, design, critique, iterate upon, and launch the user interface and user flow of applications on VA.gov.
Design General Principles
1.	Always Allow for flexibility
•	Provide choices in features and ways that tasks can be accomplished
•	Accommodate right- and left-handed use
•	Allow the user to customize settings whenever possible
•	Keep in mind that people may be using adaptive technologies
2.	Simple and intuitive
•	Do not design something differently from user expectations just to be different
•	Eliminate unnecessary complexity
•	Provide feedback
3.	Minimize errors
•	Provide warnings
•	Build fail-safe features when possible
•	Do not establish patterns when you want people to pay attention
4.	Include redundant modalities
•	using both a beep and a menu bar flash to notify a user of an error
•	using text to label images
•	redundantly allowing a user to issue commands by typing or selecting something with a pointer

**Comment by k80bowman** [comment]
This is great @CourtneyChambers77, thank you! I'll create some tickets for handling these recommended updates.
